src/io_worker.cpp

```diff
@@ -135,38 +135,39 @@ void IOWorker::add_pool(const Host::ConstPtr& host, bool is_initial_connection)
   }
 }
 
-bool IOWorker::execute(RequestHandler* request_handler) {
-  return request_queue_.enqueue(request_handler);
-}
-
-void IOWorker::retry(RequestHandler* request_handler) {
-  Address address;
-  if (!request_handler->get_current_host_address(&address)) {
-    request_handler->on_error(CASS_ERROR_LIB_NO_HOSTS_AVAILABLE,
-                              "All hosts in current policy attempted "
-                              "and were either unavailable or failed");
-    return;
+bool IOWorker::execute(const RequestHandler::Ptr& request_handler) {
+  request_handler->inc_ref(); // Queue reference
+  if (!request_queue_.enqueue(request_handler.get())) {
+    request_handler->dec_ref();
```

Comment:
I think I'd like to have a longer discussion about this later; this is a broader question that isn't particularly related to speculative execution, so I'm cool with what you've got here.